app/jobs/concerns/maintenance_tasks/task_job_concern.rb

```diff
@@ -75,6 +76,9 @@ def build_enumerator(_run, cursor:)
         MSG
       end
 
+      unless @collection_enum.is_a?(JobIteration.enumerator_builder::Wrapper)
```

Comment:
Is there a risk of `Wrapper` constant not existing in `JobIteration.enumerator_builder` namespace?